src/backend/marsha/bbb/permissions.py

```diff
@@ -59,65 +32,96 @@ def has_permission(self, request, view):
         """
         if not request.resource:
             return False
-        try:
-            return str(view.get_related_object().classroom.id) == request.resource.id
-        except ObjectDoesNotExist:
-            return False
 
+        return str(view.get_related_classroom_id()) == request.resource.id
+
+
+class BaseIsRelatedClassroomPlaylistRoleMixin:
+    """Allow request when the user has specific role on the object's classroom's playlist."""
+
+    # pylint: disable=unused-argument
+    def get_playlist_id(self, request, view, obj):
+        """Get the playlist id."""
+        # Note, use select_related to avoid making extra requests to get the video
+        return obj.classroom.playlist_id
 
-class IsClassroomPlaylistOrOrganizationAdmin(permissions.BasePermission):
+
+class BaseIsRelatedClassroomPlaylistRole(
+    BaseIsRelatedClassroomPlaylistRoleMixin, core_permissions.BaseIsObjectPlaylistRole
+):
+    """Allow request when the user has specific role on the object's classroom's playlist."""
+
+
+class IsRelatedClassroomPlaylistAdminOrInstructor(
+    core_permissions.HasAdminOrInstructorRoleMixIn,
+    BaseIsRelatedClassroomPlaylistRole,
+):
     """
-    Allow a request to proceed. Permission class.
+    Allow request when the user has admin or instructor role on the object's classroom's playlist.
+    """
+
+
+class BaseIsRelatedClassroomPlaylistOrganizationRole(
+    BaseIsRelatedClassroomPlaylistRoleMixin,
+    core_permissions.BaseIsPlaylistOrganizationRole,
+):
+    """Base permission class for object's video's playlist's organization roles."""
+
+
+class IsRelatedClassroomOrganizationAdmin(
+    core_permissions.HasAdminRoleMixIn,
+    BaseIsRelatedClassroomPlaylistOrganizationRole,
+):
+    """
+    Permission class to check if the user is one of
+    the object's video's playlist's organization admin.
```

Comment:
In this file you are use the word "video" where "classroom" could be use ? is it wanted ?

Reply:
well spot, is it not. I will fix it. Thanks